runner.py

```diff
@@ -0,0 +1,134 @@
+import runner_run_funcs
+
+SIM_TYPE_FED_LEARNING = "fed_learning"
+SIM_TYPE_FED_AVG_WITH_DP = "fed_avg_with_dp"
+
+DATA_GEN_TYPE_DATA_FROM_FILE = "file_data"
+DATA_GEN_TYPE_BLOB = "data_gen_blob"
+DATA_GEN_TYPE_RAND = "data_gen_rand"
+
+P_KEY_DATA_FILE_PATH = "data_file_path"
+
+P_KEY_NUM_SAMPLES = "num_samples"
+P_KEY_NUM_LABELS = "num_labels"
+P_KEY_NUM_FEATURES = "num_features"
+P_KEY_NUM_USERS = "num_users"
+
+P_KEY_NUM_ROUNDS = "num_rounds"
+P_KEY_BATCH_SIZE = "batch_size"
+P_KEY_NUM_EPOCHS = "num_epochs"
+
+
+class RunFuncAndReqParams:
+    def __init__(self, run_func, prereq_check_func):
+        self.run_func = run_func
+        self.prereq_params = prereq_check_func
+
+
+run_func_ltable = {
+    SIM_TYPE_FED_LEARNING: RunFuncAndReqParams(
+        runner_run_funcs.run_fed_learn_sim,
+        {
+            P_KEY_NUM_ROUNDS,
+            P_KEY_BATCH_SIZE,
+            P_KEY_NUM_EPOCHS,
+            P_KEY_NUM_SAMPLES,
+            P_KEY_NUM_FEATURES,
+            P_KEY_NUM_USERS,
+        },
+    ),
+    SIM_TYPE_FED_AVG_WITH_DP: RunFuncAndReqParams(
+        runner_run_funcs.run_fed_avg_with_dp, {P_KEY_NUM_LABELS, P_KEY_NUM_FEATURES}
+    ),
+    DATA_GEN_TYPE_DATA_FROM_FILE: RunFuncAndReqParams(
```

Comment:
Can we split run_func_ltable into two separate lists: 

  * (model, required params) 
  * (data generator, labels)

Reply:
So... I just need some clarification here.

From our meeting, I know we want the way the simulator gets data to feel more "natural" in terms of how this algo. would run in the real world. But I'm still not totally sure what this would look like. At this point should we still even support data generation right in the runner, or only allow reading in from a file?

I think in the example above  `(data generator, labels)` `data generator` is the run function to generate/read-in the data. Is this right? And would `labels` just be the number of labels in the data?

Reply:
re `(data generator, labels)` - I think so - it's hard to get an idea for the ergonomics of what the API is going to be like without iterating a bunch.  Let's just split the monolithic list for now so that we have smaller parts that are easier to test and instrument so that we can iterate 'cheaply' in terms of effort while we figure out how we want to make this work.

Reply:
Ok I'll give this a try